src/main/scala/rocket/ALU.scala

```diff
@@ -41,21 +56,22 @@ class ALUFN {
 
   def isMulFN(fn: UInt, cmp: UInt) = fn(1,0) === cmp(1,0)
   def isSub(cmd: UInt) = cmd(3)
-  def isCmp(cmd: UInt) = cmd >= FN_SLT
+  def isCmp(cmd: UInt) = (cmd >= FN_SLT && cmd <= FN_SGEU)
+  def isMaxMin(cmd: UInt) = (cmd >= FN_MAX && cmd <= FN_MINU)
```

Comment:
I'm sightly nervous on these code, maybe we can believe synthesis code? But I personally think we can decode to a onehot signal at the decode stage(but the uop will increase). This is an affordable trade off.

Reply:
This is only a handful of gates on `io.fn`, its just a bitmask ignoring the lower 2 bits of `io.fn`.

Reply:
I think using `cmd === FN_MAXMIN` at `isMaxMin` will make it easier to maintain. We can also apply this to `isCmp`.

Reply:
You mean use `BitPat` to do this?

Reply:
> You mean use `BitPat` to do this?

Yes.

Reply:
In the future, with our new Rocket framework, we'd like to use our new [rvdecoderdb](https://github.com/chipsalliance/rvdecoderdb) infra for this to handle the control signal generation, and they can be used for multiple-stage uop decoding.